app/models/task.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import String, DateTime
+from datetime import datetime
+from typing import Optional
+from sqlalchemy import ForeignKey
 from ..db import db
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] 
+    description: Mapped[str]
+    completed_at: Mapped[datetime | None] = mapped_column(nullable=True)
+
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"), nullable=True)
```

Comment:
Since you have `Optional` here we can remove the `nullable=True` from `mapped_column`.
```suggestion
goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id")
```